src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -266,6 +262,9 @@
     (.close ds))
   (when-let [pool (:job-pool context)]
     (stop-and-reset-pool! pool))
+  (when-let [command-load-future (:command-load-future context)]
+    (when-not (.isDone command-load-future)
```

Comment:
We can probably omit this guard since I think future-cancel should just return false if it's already finished.